src/ingameWindows/iwMapGenerator.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2005-2010 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Siedler II.5 RTTR.
+//
+// Siedler II.5 RTTR is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Siedler II.5 RTTR is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Siedler II.5 RTTR. If not, see <http://www.gnu.org/licenses/>.
+
+#include "defines.h" // IWYU pragma: keep
+#include "iwMapGenerator.h"
+
+#include "Loader.h"
+#include "controls/ctrlOptionGroup.h"
+#include "controls/ctrlComboBox.h"
+#include "controls/ctrlProgress.h"
+
+#include "gameData/const_gui_ids.h"
+#include "gameData/MaxPlayers.h"
+#include "helpers/containerUtils.h"
+#include "libutil/src/colors.h"
+
+#include <boost/format.hpp>
+#include <string>
+
+enum
+{
+    CTRL_PLAYER_NUMBER = 2,
```

Comment:
I assigned the values to be able to use the enums for the controls in all places, including the constructor where the controls are added (and require an ID). If I want to avoid the values, then I need to add an enum-value for all controls (even labels, which i never access again).